### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/rancher-dashboard/commit/12d77dec25f7798512ad78b96842e6142471ebc4#commitcomment-64690613

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/rancher-dashboard/commit/60c5288c5be2e478b86e6a671d6f8043b26f2ebf

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

I intend to annotate this PR shortly. I'm happy to drop changes to files, drop specific terms, or split things as requested, although this may take some time.

### CODE REVIEW

File: chart/monitoring/ClusterSelector.vue
Comment (jsoref) [inline chart/monitoring/ClusterSelector.vue:116]: brand

File: components/MessageLink.vue
Comment (jsoref) [inline components/MessageLink.vue:14]: api break?

File: components/SimpleBox.vue
Comment (jsoref) [inline components/SimpleBox.vue:22]: possibly dangerous?

File: components/dialog/harvester/SupportBundle.vue
Comment (jsoref) [inline components/dialog/harvester/SupportBundle.vue:11]: api break?

File: components/form/ChangePassword.vue
Comment (jsoref) [inline components/form/ChangePassword.vue:45]: Sometimes I'd adjust whitespace, but since `pIsRandomGenerated` is out of sync, I didn't.

File: components/form/Probe.vue
Comment (jsoref) [inline components/form/Probe.vue:194]: This is the corresponding change. But I still don't really know what it is

File: components/graph/LinePlot.vue
Comment (jsoref) [inline components/graph/LinePlot.vue:36]: api break?

File: config/labels-annotations.js
Comment (jsoref) [inline config/labels-annotations.js:146]: api break?

File: config/product/istio.js
Comment (jsoref) [inline config/product/istio.js:43]: api break?

File: config/product/istio.js
Comment (jsoref) [inline config/product/istio.js:56]: api break?

File: config/product/logging.js
Comment (jsoref) [inline config/product/logging.js:317]: ??

File: config/product/manager.js
Comment (jsoref) [inline config/product/manager.js:141]: ??

File: config/types.js
Comment (jsoref) [inline config/types.js:146]: brand?

File: edit/kubevirt.io.virtualmachine/VirtualMachineNetwork/base.vue
Comment (jsoref) [inline edit/kubevirt.io.virtualmachine/VirtualMachineNetwork/base.vue:110]: ?????

File: edit/management.cattle.io.clusterroletemplatebinding.vue
Comment (jsoref) [inline edit/management.cattle.io.clusterroletemplatebinding.vue:31]: ??

File: edit/kubevirt.io.virtualmachine/index.vue
Comment (jsoref) [inline edit/kubevirt.io.virtualmachine/index.vue:324]: api break?

File: edit/monitoring.coreos.com.prometheusrule/AlertingRule.vue
Comment (jsoref) [inline edit/monitoring.coreos.com.prometheusrule/AlertingRule.vue:16]: local?

File: edit/persistentvolume/plugins/awsElasticBlockStore.vue
Comment (jsoref) [inline edit/persistentvolume/plugins/awsElasticBlockStore.vue:36]: api?

File: edit/persistentvolumeclaim.vue
Comment (jsoref) [inline edit/persistentvolumeclaim.vue:19]: scary?

File: edit/workload/Upgrading.vue
Comment (jsoref) [inline edit/workload/Upgrading.vue:38]: local

File: list/catalog.cattle.io.app.vue
Comment (jsoref) [inline list/catalog.cattle.io.app.vue:39]: ??

File: list/harvesterhci.io.cloudtemplate.vue
Comment (jsoref) [inline list/harvesterhci.io.cloudtemplate.vue:51]: api?

File: list/harvesterhci.io.dashboard/HarvesterUpgrade.vue
Comment (jsoref) [inline list/harvesterhci.io.dashboard/HarvesterUpgrade.vue:46]: both spellings are used in this repo. quite likely an api break.

File: list/harvesterhci.io.dashboard/index.vue
Comment (jsoref) [inline list/harvesterhci.io.dashboard/index.vue:236]: api break?

File: list/harvesterhci.io.dashboard/index.vue
Comment (jsoref) [inline list/harvesterhci.io.dashboard/index.vue:246]: api break?

File: list/harvesterhci.io.setting.vue
Comment (jsoref) [inline list/harvesterhci.io.setting.vue:33]: scary, but maybe it's a local?

File: list/harvesterhci.io.setting.vue
Comment (jsoref) [inline list/harvesterhci.io.setting.vue:126]: api?

File: list/provisioning.cattle.io.cluster.vue
Comment (jsoref) [inline list/provisioning.cattle.io.cluster.vue:69]: brand

File: mixins/harvester-vm/index.js
Comment (jsoref) [inline mixins/harvester-vm/index.js:611]: ??

File: models/harvester/harvesterhci.io.setting.js
Comment (jsoref) [inline models/harvester/harvesterhci.io.setting.js:31]: api?

File: models/harvester/harvesterhci.io.setting.js
Comment (jsoref) [inline models/harvester/harvesterhci.io.setting.js:51]: api?

File: models/management.cattle.io.roletemplate.js
Comment (jsoref) [inline models/management.cattle.io.roletemplate.js:65]: api?

File: models/management.cattle.io.roletemplate.js
Comment (jsoref) [inline models/management.cattle.io.roletemplate.js:188]: api?

https://prometheus.io/docs/introduction/faq/#what-is-the-plural-of-prometheus

File: models/management.cattle.io.roletemplate.js
Comment (jsoref) [inline models/management.cattle.io.roletemplate.js:214]: api?

File: models/rke-machine.cattle.io.digitaloceanmachinetemplate.js
Comment (jsoref) [inline models/rke-machine.cattle.io.digitaloceanmachinetemplate.js:3]: api?

File: pages/home.vue
Comment (jsoref) [inline pages/home.vue:61]: ??

File: pages/support/index.vue
Comment (jsoref) [inline pages/support/index.vue:12]: I'm used to variations, I'm not used to this one, but a colleague claimed familiarity w/ it.

File: plugins/js-yaml.js
Comment (jsoref) [inline plugins/js-yaml.js:3]: brand, in fact, it should be `CodeMirror`

File: store/type-map.js
Comment (jsoref) [inline store/type-map.js:1369]: For changes this large, I don't enforce en-* variations. But as the sponsor is SUSE, I'd assume en-GB is the preferred flavor.

File: test/utils/array.js
Comment (jsoref) [inline test/utils/array.js:148]: both were used in the repo

File: assets/translations/en-us.yaml
Comment (jsoref) [reply assets/translations/en-us.yaml:5743]: ok, this was just Google Sheets suggesting changing `lifecycle` to `life cycle` and me not correctly responding to its suggestion. I'm going to drop this.

File: utils/computed.js
Comment (rak-phillip) [inline utils/computed.js:26]: Will this change be fine if we're changing the name of a positional argument? 

I think it will be, but I'm leaving this note as a reminder to double-check.

File: test/utils/array.js
Comment (rak-phillip) [reply test/utils/array.js:148]: Either is valid, but falsy is more common from my experience.

File: store/type-map.js
Comment (rak-phillip) [reply store/type-map.js:1369]: Your reasoning is sound. I've always left it as en-GB when written as such. We do tend to weave back and forth depending on author 🙂

@richard-cox do we have a preference that we enforce, or is it truly up to the author?

File: plugins/steve/resource-class.js
Comment (rak-phillip) [inline plugins/steve/resource-class.js:1340]: This one will require closer review, as we're modifying an object property and need to name sure that the corresponding change `metdata` => `metadata` is in place.

File: pages/support/index.vue
Comment (rak-phillip) [reply pages/support/index.vue:12]: `REGEX` is definitely more common in my experience.

File: pages/home.vue
Comment (rak-phillip) [reply pages/home.vue:61]: Marking this one to come back to for closer review.

File: pages/c/_cluster/explorer/tools/index.vue
Comment (rak-phillip) [inline pages/c/_cluster/explorer/tools/index.vue:214]: Changing route params here will require closer inspection.

File: models/rke-machine.cattle.io.digitaloceanmachinetemplate.js
Comment (rak-phillip) [reply models/rke-machine.cattle.io.digitaloceanmachinetemplate.js:3]: Modifying an export - commenting to circle back for a closer look later on.

Comment (rak-phillip) [reply models/rke-machine.cattle.io.digitaloceanmachinetemplate.js:3]: With that said, it is a default export and not a named export, so I think this change is safe..

File: models/management.cattle.io.roletemplate.js
Comment (rak-phillip) [reply models/management.cattle.io.roletemplate.js:188]: `Prometheuses` is intentional - [see this comment](https://github.com/rancher/dashboard/pull/4282#issue-1006795952)

File: mixins/harvester-vm/index.js
Comment (rak-phillip) [reply mixins/harvester-vm/index.js:611]: I expect that this would always evaluate `false`..

File: list/harvesterhci.io.cloudtemplate.vue
Comment (rak-phillip) [reply list/harvesterhci.io.cloudtemplate.vue:51]: Computed prop for this component, this change is safe

File: list/catalog.cattle.io.app.vue
Comment (rak-phillip) [reply list/catalog.cattle.io.app.vue:39]: `class` is the right choice here

File: edit/persistentvolume/plugins/awsElasticBlockStore.vue
Comment (rak-phillip) [reply edit/persistentvolume/plugins/awsElasticBlockStore.vue:36]: computed prop - safe to change.

File: edit/monitoring.coreos.com.prometheusrule/AlertingRule.vue
Comment (rak-phillip) [reply edit/monitoring.coreos.com.prometheusrule/AlertingRule.vue:16]: yes, looks safe to change

File: edit/management.cattle.io.clusterroletemplatebinding.vue
Comment (rak-phillip) [reply edit/management.cattle.io.clusterroletemplatebinding.vue:31]: `errors` is correct

File: edit/kubevirt.io.virtualmachine/index.vue
Comment (rak-phillip) [reply edit/kubevirt.io.virtualmachine/index.vue:324]: component method - looks safe to change

File: edit/kubevirt.io.virtualmachine/VirtualMachineNetwork/base.vue
Comment (rak-phillip) [reply edit/kubevirt.io.virtualmachine/VirtualMachineNetwork/base.vue:110]: computed prop - safe to change

**Review by rak-phillip — COMMENTED** [review]
Thanks for the PR 🙂

I'm about halfway through and will pick up more later. I might enlist some help from the likes of @richard-cox (time permitting) due to the scope of these changes.

File: mixins/harvester-vm/index.js
Comment (jsoref) [reply mixins/harvester-vm/index.js:611]: I guess that means that if you have `userDataJson = {ssh_authorized_keys: []}` then it'll stay that way instead of having the property removed.

I suppose there could be code that isn't properly assuming the field might not be present...

**Review by richard-cox — CHANGES_REQUESTED** [review]
In previous projects we've had problems with generic tools making changes without understanding the context of them. I'm concerned that the spelling changes to code will have knock-on effects. Ideally maybe these changes should just apply to user facing values in both *.yaml files

Additionally
- The 50 commits should be squashed in to one
- There are a lot of changes with comments such as `scary`, `??????`, `??`, `api break?`, etc. Can they be resolved by action or comment expanded?

**Comment by jsoref** [comment]
Happy to squash commits at the end. But I don't want to do it until then.

If a given change turns out to be objectionable it's very hard for me to safely remove it once I've squashed the changes together.

I'm a bit confused by the second bullet. I have a lot of experience working on large code bases. I have a general sense of which classes of changes tend to be risky, so I try to help reviewers by calling things out. For some comments, @rak-phillip has replied indicating they aren't problematic, so I've marked them as resolved. The review isn't complete, so some aren't resolved, and I'm not going to apply a reply from one comment to another just because I'm fairly certain they're the same category. I'm not in any hurry (Ask @mook-as ).

Do note that some of the code changes very much apply to bugs in the code as opposed to merely user visible text. This is absolutely intentional. My background is as a developer where I regularly run into logic bugs because of such typos. A couple of these have been called out here and again in some cases, @rak-phillip has already indicated they're ok. In at least one case I've replied describing what I think the risk is even though the logic change should be correct, in case someone else wants to check callers to see if they're fragile/at risk to the behavior change.

**Comment by richard-cox** [comment]
My concern is the high level of risk making blind, contextless changes to all parts of the repo that need to be fully understood and tested before merging (which has a cost). It would make more sense making targeted changes to specific translations, which is pretty straight forward with a spell checking plug-in for vs code. 

In code I think there was a place where `length` was misspelt, it would be easier to pull that fix out than to validate all other code changes. In the future when we transition to typescript those kind of errors will be handled much much earlier.

As it stands I'm bumping this out to `2.6.5` to ensure all committed `2.6.4` items can be delivered.

**Comment by jsoref** [comment]
I'm quite happy to split out the ones that are code fixes. Or all the ones that are translation items. Any split like that isn't a big deal. I just need a direction.

And yes, I'm quite happy with not rushing. While the PR(s) has(/ve) individual commits, rebasing to deal w/ conflicts isn't a big deal.

File: plugins/steve/resource-class.js
Comment (jsoref) [reply plugins/steve/resource-class.js:1340]: I've tried searching a couple of ways, and I can't find this _anywhere_ ...

There are 3 copies of this file in org:rancher
https://github.com/search?q=org%3Arancher+metdata&type=code
https://cs.github.com/?scopeName=All+repos&scope=&q=org%3Arancher+metdata

metadata-store is also occasionally misspelled:
https://cs.github.com/?scopeName=All+repos&scope=&q=rancher+metdata

But... not really sure how else to search...

File: utils/computed.js
Comment (jsoref) [reply utils/computed.js:26]: JavaScript isn't python, so there's no concept of targeting a named argument.

A method call fills argument slots, and a method definition retrieves from those slots. So this is harmless.

Effectively `delimiter` is the local alias for `arguments[1]` (`0`-based).

<details><summary>Node</summary>

```node
// Welcome to Node.js v17.3.1.
> let a=function a(b,c='2') {console.log(b,c)}
undefined
> a(1,4)
1 4
undefined
> a(1,e=4)
1 4
undefined
> (function () { console.log(arguments.length)})()
0
undefined
> (function () { console.log(arguments.length)})(1)
1
undefined
> (function () { console.log(arguments.length)})(1,2)
2
undefined
> (function () { console.log(arguments[1])})(1,2)
2
undefined
> (function () { console.log(arguments[1])})(1,3,2)
3
undefined
```
</details>

File: models/rke-machine.cattle.io.digitaloceanmachinetemplate.js
Comment (jsoref) [reply models/rke-machine.cattle.io.digitaloceanmachinetemplate.js:3]: GitHub tells me that it only offered one other instance of this string:
https://cs.github.com/?scopeName=All+repos&scope=&q=DigialoceanMachineTemplate
https://github.com/bart-rijnders/dashboard/blob/21c71f84cc81342a86d116a7ccef88cfee450027/models/rke-machine.cattle.io.digitaloceanmachinetemplate.js#L3
That's because https://github.com/bart-rijnders/dashboard -- This branch is 622 commits behind rancher:master.

**Comment by rak-phillip** [comment]
> I'm quite happy to split out the ones that are code fixes. Or all the ones that are translation items. Any split like that isn't a big deal. I just need a direction.
> 
> And yes, I'm quite happy with not rushing. While the PR(s) has(/ve) individual commits, rebasing to deal w/ conflicts isn't a big deal.

@jsoref considering the scope of these changes, it'll be best start with the low hanging fruit like translations, comments, etc... and raise new PRs for any changes that might affect code and logic. 

We might need to distill down future PRs further to limit the scope of testing depending on the risk associated with the code changes, but I think getting these low risk corrections out of the way will be a great improvement to start with.

File: store/type-map.js
Comment (catherineluse) [reply store/type-map.js:1369]: The documentation uses American spelling. For a while it was kind of Western US themed and said things like "Howdy pardner."

File: assets/translations/en-us.yaml
Comment (rak-phillip) [reply assets/translations/en-us.yaml:872]: This usage has been replaced in `components/form/ChangePassword.vue`

File: assets/translations/en-us.yaml
Comment (rak-phillip) [reply assets/translations/en-us.yaml:3280]: Looks good as this is just alternate placeholder text for an input.

File: assets/translations/en-us.yaml
Comment (rak-phillip) [reply assets/translations/en-us.yaml:4536]: Looks good

File: assets/translations/zh-hans.yaml
Comment (rak-phillip) [reply assets/translations/zh-hans.yaml:177]: I think keeping it consistent with `en-us` is best here, as this hasn't been translated to `zh-hans` yet.

File: components/dialog/harvester/SupportBundle.vue
Comment (rak-phillip) [reply components/dialog/harvester/SupportBundle.vue:11]: Looks good - I don't see the component being referenced by name with the misspelled name.

File: components/form/Probe.vue
Comment (rak-phillip) [reply components/form/Probe.vue:194]: Just a second placeholder, that's all.

File: config/product/logging.js
Comment (rak-phillip) [reply config/product/logging.js:318]: `array[string]` is the proper type.. the variation with `rrr` also isn't used anywhere else, telling me this is a typo

File: store/type-map.js
Comment (rak-phillip) [reply store/type-map.js:1369]: Thanks @catherineluse 🙂

I'm not perturbed about the variation for this one. Seeing as how it's a code comment, I'm a little inclined to just leave it as `en-gb`.

**Review by rak-phillip — APPROVED** [review]
LGTM 

@jsoref thanks for paring this PR down for us, it's much appreciated 😁

@richard-cox regarding the concerns about the number of commits, I think it would suffice for us to squash on merge through github. Please let me know if there's any concern with taking that approach.

@catherineluse @richard-cox All the changes look safe to me, but I think a second pair of eyes would be good on this one (time permitting), just in case I missed something.

File: store/type-map.js
Comment (jsoref) [reply store/type-map.js:1369]: If the formal style is `en-US`, I'm happy to make a distinct PR for the repo (or repos) to switch. cowboy style language is fine, but I won't cowboy it, it'd go in as a distinct PR.

**Comment by jsoref** [comment]
From experience, generally a squash merge of a PR (that I make) like this is fine.

I'd encourage you to prune the commit message that it writes, the default message will have 64 copies of `signed-off-by` which is pretty unhelpful.

My default commit message is something of the form:

```
git commit -m "Spelling fixes #4994

messages, translations, comments, internal elements, seemingly harmless items

$(git log master..spelling|perl -ne 'next unless s/.*spelling:/*/;print'|tac)

Signed-off-by: Josh Soref <jsoref@users.noreply.github.com>"
```

Something like that... adjust to taste. -- The `*` on the second side of the `s///` is a Markdown bullet.

GitHub will forever remember the branch you've merged as `refs/pull/4994/head`, which means if one wants to back things out, one can retrieve the individual commits from that branch.

File: store/type-map.js
Comment (richard-cox) [reply store/type-map.js:1369]: As much as it pains me to say... `en-US` should be used everywhere. It's not critical that we fix existing text though

**Review by richard-cox — APPROVED** [review]

**Comment by richard-cox** [comment]
FWIW i've created https://github.com/rancher/dashboard/pull/5100 to cover three of the code changes. I've tested one and requested the other two checked by someone on the Harvester team

**Pull request merged**
